index.js

```diff
@@ -20,35 +20,45 @@ const exportDefaultFrom = require('@babel/plugin-proposal-export-default-from').
 const proposalOptionalChaining = require('@babel/plugin-proposal-optional-chaining').default;
 const removePropTypes = require('babel-plugin-transform-react-remove-prop-types').default;
 
-const { browserlist, legacyBrowserList } = require('./browserlist');
-
-module.exports = () => ({
-  env: {
-    modern: {
-      presets: [
-        [envPreset, {
-          targets: {
-            browsers: browserlist,
-            node: '12.16.1',
-          },
-        }],
+const { browserList, legacyBrowserList } = require('./browserlist');
+
+module.exports = (api = {}, opts = {}) => {
+  const serverOnly = opts.serverOnly || (api.env && api.env('server'));
+  const isModern = opts.modern || (api.env && api.env('modern'));
+
+  const targets = {
+    node: 'current',
+  };
+
+  if (!serverOnly) {
+    targets.browsers = isModern ? browserList : legacyBrowserList;
+  }
+
+  const presetEnvOptions = Object.assign(
+    {},
+    { targets },
```

Comment:
```suggestion
    {
      // https://babeljs.io/docs/en/babel-preset-env#spec
      spec: true,
      // preserve ES syntax and allow webpack / rollup to handle the transformed output
      // https://babeljs.io/docs/en/babel-preset-env#modules
      modules: false,
      // https://babeljs.io/docs/en/babel-preset-env#bugfixes
      bugfixes: isModern,
      // https://babeljs.io/docs/en/babel-preset-env#corejs
      corejs: 3,
    },
```

Reply:
Sane defaults? Setting `modules` to `false` will work better with our bundlers by preserving ES syntax. `bugifxes` will remove extra code from modern builds, `spec` will up our compatibility I believe and we set the corejs version to be explicit.

Reply:
Not sure why we should remove targets, the purpose is to determine the best configuration for your indented use. 

`spec: true,`  Appears to have no effect on the server only builds. Might increase compliance but at the cost of speed.
> Enable more spec compliant, but potentially slower, transformations 

`modules: false,` Would be break every build, including one app, specific use cases where no transformation is required can configure.

`corejs: 3,` `useBuiltIns` is not specified so this option has no effect. 

`bugfixes: isModern,` might be worth including however i am not seeing any change in build output, i suspect the use of targets negates the need for configuring this